lib/opam_solve.ml

```diff
@@ -90,12 +90,22 @@ module Opam_monorepo_context (Base_context : BASE_CONTEXT) :
             else (version, Error Non_dune))
       versions
 
+  let sort_candidates versions =
```

Comment:
I am not sure "sort" is the right term for it, given it doesn't really sort them but rather reorders them.

Reply:
Maybe `demote_candidates_to_avoid`? I would say it doesn't really sort them (because it requires the candidates to be sorted, otherwise we could conclude that the identity function is a sorting function that just requires the the input to be sorted), it more splits or partitions them.

Reply:
Good name indeed, I'll change it!